src/machines/appMachine.ts

```diff
@@ -0,0 +1,57 @@
+import { ActorRefFrom, assign, createActor, setup, spawnChild } from 'xstate'
+import { authMachine } from './authMachine'
+import { useSelector } from '@xstate/react'
+import { ACTOR_IDS } from './machineConstants'
+import { settingsMachine } from './settingsMachine'
+import { createSettings } from 'lib/settings/initialSettings'
+
+const { AUTH, SETTINGS } = ACTOR_IDS
+const appMachineActors = {
+  [AUTH]: authMachine,
+  [SETTINGS]: settingsMachine,
+} as const
+
+type AppMachineActors = {
+  [K in keyof typeof appMachineActors]: ActorRefFrom<
+    (typeof appMachineActors)[K]
+  >
+}
+
+const appMachine = setup({
+  actors: appMachineActors,
+}).createMachine({
+  /** @xstate-layout N4IgpgJg5mDOIC5gF8A0IB2B7CdGgAoBbAQwGMALASwzAEp8QAHLWKgFyqw0YA9EAjACZ0AT0FDkU5EA */
+  id: 'modeling-app',
+  entry: [
+    spawnChild(AUTH, { id: AUTH, systemId: AUTH }),
+    spawnChild(SETTINGS, {
+      id: SETTINGS,
+      systemId: SETTINGS,
+      input: createSettings(),
+    }),
+  ],
+})
+
+export const appActor = createActor(appMachine)
+export const authActor = appActor.system.get(AUTH) as ActorRefFrom<
+  typeof authMachine
+>
```

Comment:
XState wasnt able to infer on its own? (just curious)

Reply:
No! This honestly is so frustrating to me. I figure it's just this one POS line for now but I do not want to have to do this for every actor in the system. I want to find a better way but I was just getting `any` back from `system.get()`. Maybe there's a type parameter I should add?

Reply:
Not a clue. We'll have to look together more closely but isn't a blocker for me.